Amazon.QLDB.Driver/driver/QldbDriver.cs

```diff
@@ -14,31 +14,47 @@
 namespace Amazon.QLDB.Driver
 {
     using System;
+    using System.Collections.Concurrent;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Threading;
+    using Amazon.QLDBSession;
+    using Amazon.QLDBSession.Model;
     using Amazon.Runtime;
+    using Microsoft.Extensions.Logging;
 
     /// <summary>
     /// <para>Represents a factory for accessing a specific ledger within QLDB. This class or
     /// <see cref="AsyncQldbDriver"/> should be the main entry points to any interaction with QLDB.</para>
     ///
-    /// <para>This factory pools sessions and attempts to return unused but available sessions when getting new sessions.
+    /// <para>
+    /// This factory pools sessions and attempts to return unused but available sessions when getting new sessions.
     /// The pool does not remove stale sessions until a new session is retrieved. The default pool size is the maximum
     /// amount of connections the session client allows set in the <see cref="ClientConfig"/>. <see cref="Dispose"/>
-    /// should be called when this factory is no longer needed in order to clean up resources, ending all sessions in the pool.</para>
+    /// should be called when this factory is no longer needed in order to clean up resources, ending all sessions in
+    /// the pool.
+    /// </para>
     /// </summary>
     public class QldbDriver : BaseQldbDriver, IQldbDriver
     {
-        private readonly SessionPool sessionPool;
+        private readonly BlockingCollection<QldbSession> sessionPool;
```

Comment:
Can this exist in the base and you templatize BaseQldbDriver with the type of element going into the collection?

Reply:
No it will not work. QldbSession is an internal class so you cannot use it in a publicly exposed implementation type (BaseQldbDriver<QldbSession>.)